lib/Net/SIP/Dispatcher.pm

```diff
@@ -861,24 +861,36 @@ sub __resolve_uri_final {
     # FIXME: can contradict order in @proto
     @$resp = sort { $a->{prio} <=> $b->{prio} } @$resp;
 
+    # remove duplicate entries in response
+    my %resp;
+    for(my $i=0; $i<@$resp; $i++) {
+	my $r = $resp->[$i];
+	if (exists $resp{$r->{host}}->{$r->{family}}->{$r->{proto}}->{$r->{addr}}->{$r->{port}}) {
+	    splice(@$resp,$i,1);
```

Comment:
Since the current element will be removed by this `$i` needs to be updated to reflect this, so that `$i++` in the for loop will only point to the next element (which is now at position `$i` and not `$i+1`) and not to the one after that.

Reply:
You are right! This is an issue, I will fix it.

Reply:
Fixed.